### PR TITLE
Add support for Symfony 6

### PR DESCRIPTION
**Pull request by loic425**

### CODE REVIEW

**Review by ciaranmcnulty — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,6 +28,8 @@ jobs:
                         composer-extras: "symfony/event-dispatcher ^3.4 symfony/contracts ^2.0"
                     -   php: 8.0
                         operating-system: windows-latest
+                        stability: dev
+                        SYMFONY_REQUIRE: 6.0.*
```
[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:33]: oups, my bad, that was a  mistake.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -41,14 +43,17 @@ jobs:
                     php-version: "${{ matrix.php }}"
                     ini-values: "phar.readonly=0"
 
+            -   name: Configure Composer minimum stability
+                if: matrix.stability
+                run: composer config minimum-stability ${{ matrix.stability }}
+
+            -   name: Install symfony/flex
```
[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:51]: Cause Symfony 6 is only available on beta versions.

[the remark this answers is not in the document]
Comment (ciaranmcnulty) [reply .github/workflows/build.yml:51]: So we need Flex?

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:51]: Flex is use to restrict Symfony versions. Maybe I forgot one thing. I am checking that.

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:51]: No that's good I added the SYMFONY_REQUIRE env var on install composer step.
please look at https://github.com/Behat/Behat/blob/master/.github/workflows/build.yml

[the remark this answers is not in the document]
Comment (ciaranmcnulty) [reply .github/workflows/build.yml:51]: I'm not sure how the Flex stuff works, can you explain why I'm 5 why it's required?

I'd rather keep the CI env free of components we don't directly depend on

[the remark this answers is not in the document]
Comment (ciaranmcnulty) [reply .github/workflows/build.yml:51]: Ah I see it's used to access the env var. Can we put the deps in `composer_extras` instead?

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:51]: Hum... I think it should be setup before running the install command to restrict the Symfony packages versions.
The extra section does not update anything (--no-update option is passed). Moreover Symfony flex is added on global requirements in order to get out of your projects' dependencies.

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:51]: @nicolas-grekas Can you confirm or not please?

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:51]: Flex should be installed before the composer to restrict the Symfony versions. You passed a "--no-update" on composer extras section. Moreover here, we added the requirements on global to get rid of your projects dependencies to setup it. 
@nicolas-grekas Can you confirm or not please?

[the remark this answers is not in the document]
Comment (ciaranmcnulty) [reply .github/workflows/build.yml:51]: We run composer update later. For testing a specific version I'd rather be specific about the components than introduce Flex (which is really for symfony projects, not tools like this one that consume components)

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:51]: It's to be sure all the Symfony components you use on the CI is on the wanted version. It's not used here as a tool for Symfony projects but for the composer hook inside. You can check CI on other PHP tools using Symfony and this is setup like that.

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:51]: But I can remove that part if you prefer to.

* https://github.com/symfony/maker-bundle/blob/main/.github/workflows/ci.yaml#L88
* https://github.com/Behat/Behat/blob/master/.github/workflows/build.yml#L15

[the remark this answers is not in the document]
Comment (nicolas-grekas) [reply .github/workflows/build.yml:51]: flex is useful to control the range of symfony components, transitive deps included
I suppose that without it, composer will prefer stable versions of transitive symfony components, which is not what you want, right?

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:51]: @ciaranmcnulty Could we finish this discussion? Symfony 6.0 will be released in few days (already beta.3 has been released).

[the remark this answers is not in the document]
Comment (ciaranmcnulty) [reply .github/workflows/build.yml:51]: Can you change this step to only install Flex if we have the env var set? Then the other runs will find any hidden flex dependencies

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:52]: @ciaranmcnulty done

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:52]: @ciaranmcnulty is that ok for you?

[the remark this answers is not in the document]
Comment (loic425) [reply .github/workflows/build.yml:52]: Symfony 6.0.0 has been released today.

**Comment by loic425** [comment]
@ciaranmcnulty I can remove the Flex part If you want cause I think composer will choose the right transitive symfony components cause Symfony 6.0 has a stable release now.
Are you ok?

**Comment by loic425** [comment]
> @ciaranmcnulty I can remove the Flex part If you want cause I think composer will choose the right transitive symfony components cause Symfony 6.0 has a stable release now. Are you ok?

So I created an alternative PR without Flex dependency and without changing the CI to add "non-stable" packages support.
#1402

**Pull request closed without merging**